### PR TITLE
Config refactor

### PR DESCRIPTION
**Pull request by NeumimTo**
This is the second part of changes i promised. 

Do not merge yet, ill let you know once its ready :)

### CODE REVIEW

**Comment by NeumimTo** [comment]
Config:
```
id: "example"
name: "example"
type: "minecraft:stone_axe"
nbt: [
    {
        __class__: "com.onaple.itemizer.data.SomeNbtFactoryClass"
        value: 10
    }
]
```
Code:
```
@ConfigSerializable
public class Example implements ItemNbtFactory {
    
    @Setting("value")
    private int data;
    
    @Override
    public Key getKey() {
        return MyKeys.SOME_INT_KEY;
    }

    @Override
    public DataManipulator<?, ?> constructDataManipulator() {
        return new SomeDataManipulator(data, getKey());
    }
}
```

File: src/main/java/com/onaple/itemizer/data/serializers/MinerSerializer.java [hunk 1]
```diff
@@ -38,7 +37,7 @@ public MinerBean deserialize(TypeToken<?> type, ConfigurationNode value) throws
                 inheritances.add(inheritance);
             }
         }
-        return new MinerBean(id, mineTypes, inheritances);
+        return null;//new MinerBean(id, mineTypes, inheritances);
```
Comment (raiponz) [inline src/main/java/com/onaple/itemizer/data/serializers/MinerSerializer.java]: why always return null ? and if we always return null why we keep it ?

File: src/main/java/com/onaple/itemizer/data/serializers/PoolSerializer.java [hunk 2]
```diff
@@ -28,7 +30,7 @@ public PoolBean deserialize(TypeToken<?> type, ConfigurationNode value) throws O
                 item = ItemDAO.getItem(reference);
             }
             if (!item.isPresent() && itemType != null){
```
Comment (raiponz) [inline src/main/java/com/onaple/itemizer/data/serializers/PoolSerializer.java]: You can remove the whole condition.

File: src/main/java/com/onaple/itemizer/Itemizer.java [hunk 3]
```diff
@@ -119,9 +120,6 @@ public void preInit(GamePreInitializationEvent event) {
             Itemizer.getLogger().error(e.getMessage());
         }
         craftRegister.register(configurationHandler.getCraftList());
```
Comment (raiponz) [inline src/main/java/com/onaple/itemizer/Itemizer.java]: Crafts must be registered latest in the pre-init event

**Comment by raiponz** [comment]
Can you also solve conflict (apply changes from your side on mine, I trust you :wink: )

File: src/main/java/com/onaple/itemizer/data/serializers/MinerSerializer.java (hunk 1 above)
Comment (NeumimTo) [reply src/main/java/com/onaple/itemizer/data/serializers/MinerSerializer.java]: This entire class will be removed,

File: src/main/java/com/onaple/itemizer/data/serializers/PoolSerializer.java (hunk 2 above)
Comment (NeumimTo) [reply src/main/java/com/onaple/itemizer/data/serializers/PoolSerializer.java]: The entire class will be removed

File: src/main/java/com/onaple/itemizer/data/serializers/MinerSerializer.java (hunk 1 above)
Comment (raiponz) [reply src/main/java/com/onaple/itemizer/data/serializers/MinerSerializer.java]: Remove all that should be, don't be shy!

**Comment by Hbeau** [comment]
And Now travis is angry ! What's happened

File: src/main/java/com/onaple/itemizer/Itemizer.java [hunk 4]
```diff
@@ -274,23 +273,22 @@ private void loadGlobalConfig() {
...
     public int loadPools() throws Exception {
         initDefaultConfig("pools.conf");
-        return configurationHandler.readPoolsConfiguration(configurationHandler.loadConfiguration(configDir + "/itemizer/pools.conf"));
+        return configurationHandler.readPoolsConfiguration(configDir + "/itemizer/pools.conf");
     }
 
     public int loadCrafts() throws Exception {
         initDefaultConfig("crafts.conf");
-        return configurationHandler.readCraftConfiguration(configurationHandler.loadConfiguration(configDir + "/itemizer/crafts.conf"));
+        return configurationHandler.readCraftConfiguration(configDir + "/itemizer/crafts.conf");
```
Comment (Hbeau) [inline src/main/java/com/onaple/itemizer/Itemizer.java]: A conversion matter, I supposed you are working on it.  courage, crafts are realy complexes.

**Comment by Hbeau** [comment]
Nice job! I read that tomorrow 👍

File: src/main/java/com/onaple/itemizer/data/serializers/IngredientMapAdapter.java [hunk 5]
```diff
@@ -24,6 +25,10 @@
             String s = entry.getKey().toString();
             ConfigurationNode node = entry.getValue();
             ItemStack item = new ItemBeanRefOrItemIdAdapter().deserialize(type, node.getNode("item"));
+            if (item == null) {
+                Itemizer.getLogger().warn("Unknown item " + node.getNode("item").toString());
+                continue;
```
Comment (Hbeau) [inline src/main/java/com/onaple/itemizer/data/serializers/IngredientMapAdapter.java]: (pro tips I ve learned at work) the best way to use Slf4j is to write the message with {} in the first arg, and objects you want to log in others args. 😉

**Pull request merged**